registry/handlers/blobupload.go

```diff
@@ -241,10 +241,11 @@ func (buh *blobUploadHandler) PutBlobUploadComplete(w http.ResponseWriter, r *ht
 
 	desc, err := buh.Upload.Commit(buh, distribution.Descriptor{
 		Digest: dgst,
+		Size:   buh.State.Offset,
```

Comment:
I see. I relied on my experiments where the `buh.State.Offset` was always correct. But that's because Docker client always sends final `PUT` with no bytes to upload, which of course may change.